### PR TITLE
refactor: implement AsRef in uint.rs

### PR DESCRIPTION
**Pull request by fahimahmedx**
## Description
[AXE-2636]
Uses AsRef for Uint256

## Steps to Test
`cargo test` passes

[AXE-2636]: https://axelarnetwork.atlassian.net/browse/AXE-2636?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cgorenflo — APPROVED** [review]

**Comment by cjcobb23** [comment]
uint256 is not that small fwiw

**Review by cjcobb23 — CHANGES_REQUESTED** [review]
why are we copying uint256 now?

File: packages/axelar-wasm-std/src/nonempty/uint.rs [hunk 1]
```diff
@@ -76,12 +70,6 @@ impl From<Uint256> for cosmwasm_std::Uint256 {
     }
 }
 
-impl<'a> From<&'a Uint256> for &'a cosmwasm_std::Uint256 {
-    fn from(value: &'a Uint256) -> Self {
-        &value.0
-    }
-}
-
```
Comment (cjcobb23) [inline packages/axelar-wasm-std/src/nonempty/uint.rs:84]: Can we implement `AsRef`?

File: packages/axelar-wasm-std/src/snapshot.rs [hunk 2]
```diff
@@ -25,7 +25,7 @@ impl Snapshot {
         let participants: HashMap<String, Participant> = participants
             .into_iter()
             .map(|participant| {
-                let weight: &Uint256 = (&participant.weight).into();
+                let weight: Uint256 = participant.weight.into();
```
Comment (cjcobb23) [inline packages/axelar-wasm-std/src/snapshot.rs:28]: And then can we just use `AsRef` here?

File: packages/axelar-wasm-std/src/snapshot.rs [hunk 3]
```diff
@@ -42,10 +42,10 @@ impl Snapshot {
         }
     }
 
-    pub fn get_participant_weight(&self, participant: &Addr) -> Option<&Uint256> {
+    pub fn get_participant_weight(&self, participant: &Addr) -> Option<Uint256> {
         self.participants
             .get(participant.as_str())
-            .map(|p| (&p.weight).into())
+            .map(|p| p.weight.into())
```
Comment (cjcobb23) [inline packages/axelar-wasm-std/src/snapshot.rs:48]: And use `AsRef` here too?

**Review by cjcobb23 — APPROVED** [review]

**Pull request merged**